File: ExceptionHandler/exception.java
import java.io.*;
import java.nio.file.*;

class DivisionByZeroException extends Exception {
    public DivisionByZeroException(String message) {
        super(message);
    }
}

public class ExceptionHandling {
    
    public static double divide(double a, double b) throws DivisionByZeroException {
        if (b == 0) {
            throw new DivisionByZeroException("Division by zero is not allowed");
        }
        return a / b;
    }
    
    public static void readFile(String filename) throws IOException {
        String content = Files.readString(Paths.get(filename));
        System.out.println(content);
    }
    
    public static void demonstrateMultipleExceptions() {
        try {
            // This might throw NumberFormatException
            int number = Integer.parseInt("abc");
            
            // This might throw ArrayIndexOutOfBoundsException
            int[] array = {1, 2, 3};
            System.out.println(array[5]);
            
        } catch (NumberFormatException e) {
            System.err.println("Number format error: " + e.getMessage());
        } catch (ArrayIndexOutOfBoundsException e) {
            System.err.println("Array index error: " + e.getMessage());
        } catch (Exception e) {
            System.err.println("Unexpected error: " + e.getMessage());
        } finally {
            System.out.println("Finally block always executes");
        }
    }
    
    public static void main(String[] args) {
        // Division example with exception handling
        System.out.println("Division Examples:");
        
        try {
            double result = divide(10.0, 2.0);
            System.out.println("10 / 2 = " + result);
            
            result = divide(10.0, 0.0); // This will throw an exception
            System.out.println("10 / 0 = " + result);
            
        } catch (DivisionByZeroException e) {
            System.err.println("Error: " + e.getMessage());
        }
        
        // File operation with exception handling
        System.out.println("\nFile Operation Example:");
        try {
            readFile("nonexistent.txt");
        } catch (IOException e) {
            System.err.println("File error: " + e.getMessage());
        }
        
        // Multiple exception types
        System.out.println("\nMultiple Exception Example:");
        demonstrateMultipleExceptions();
    }
}

File: ExceptionHandler/exception.py
class DivisionByZeroError(Exception):
    """Custom exception for division by zero"""
    def __init__(self, message="Division by zero is not allowed"):
        self.message = message
        super().__init__(self.message)

def divide(a, b):
    if b == 0:
        raise DivisionByZeroError(f"Cannot divide {a} by zero")
    return a / b

def read_file(filename):
    try:
        with open(filename, 'r') as file:
            content = file.read()
        return content
    except FileNotFoundError:
        raise FileNotFoundError(f"File '{filename}' not found")
    except PermissionError:
        raise PermissionError(f"Permission denied to read '{filename}'")

def demonstrate_multiple_exceptions():
    try:
        # This might raise ValueError
        number = int("abc")
        
        # This might raise IndexError
        my_list = [1, 2, 3]
        print(my_list[5])
        
    except ValueError as e:
        print(f"Value error: {e}")
    except IndexError as e:
        print(f"Index error: {e}")
    except Exception as e:
        print(f"Unexpected error: {e}")
    else:
        print("No exceptions occurred")
    finally:
        print("Finally block always executes")

def main():
    # Division example with exception handling
    print("Division Examples:")
    
    try:
        result = divide(10, 2)
        print(f"10 / 2 = {result}")
        
        result = divide(10, 0)  # This will raise an exception
        print(f"10 / 0 = {result}")
        
    except DivisionByZeroError as e:
        print(f"Error: {e}")
    except Exception as e:
        print(f"Unexpected error: {e}")
    
    # File operation with exception handling
    print("\nFile Operation Example:")
    try:
        content = read_file("nonexistent.txt")
        print(content)
    except (FileNotFoundError, PermissionError) as e:
        print(f"File error: {e}")
    
    # Multiple exception types
    print("\nMultiple Exception Example:")
    demonstrate_multiple_exceptions()

if __name__ == "__main__":
    main()